eth-contracts/contracts/EthRewardsManager.sol

```diff
@@ -0,0 +1,165 @@
+        "EthRewardsManager: token is not a contract"
+    );
+    string private constant ERROR_WORMHOLE_NOT_CONTRACT = (
+        "EthRewardsManager: wormhole is not a contract"
+    );
+    string private constant ERROR_ONLY_GOVERNANCE = (
+        "EthRewardsManager: Only governance"
+    );
+
+    address private governanceAddress;
+
+    /// @dev ERC-20 token that will be used to stake with
+    ERC20 internal audiusToken;
+    Wormhole internal wormhole;
+    bytes32 internal recipient;
+
+    address public botOracle;
+
+    /**
+     * @notice Function to initialize the contract
+     * @param _tokenAddress - address of ERC20 token
+     * @param _governanceAddress - address for Governance proxy contract
+     */
+    function initialize(
+        address _tokenAddress,
+        address _governanceAddress,
+        address _wormholeAddress,
+        bytes32 _recipient,
+        address _botOracle
+    ) public initializer {
+        require(Address.isContract(_tokenAddress), ERROR_TOKEN_NOT_CONTRACT);
+        require(
+            Address.isContract(_wormholeAddress),
+            ERROR_WORMHOLE_NOT_CONTRACT
+        );
+        audiusToken = ERC20(_tokenAddress);
+        wormhole = Wormhole(_wormholeAddress);
+        recipient = _recipient;
+        botOracle = _botOracle;
+        _updateGovernanceAddress(_governanceAddress);
+        InitializableV2.initialize();
+    }
+
+    /**
+     * @notice Set the Governance address
+     * @dev Only callable by Governance address
+     * @param _governanceAddress - address for new Governance contract
+     */
+    function setGovernanceAddress(address _governanceAddress) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+        _updateGovernanceAddress(_governanceAddress);
+    }
+
+    /**
+     * @notice Set the recipient address
+     * @dev Only callable by Governance address
+     * @param _recipient - address for new recipient
+     */
+    function setRecipient(bytes32 _recipient) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+        recipient = _recipient;
+    }
+
+    /**
+     * @notice Set the botOracle address
+     * @dev Only callable by Governance address
+     * @param _botOracle - address for new botOracle
+     */
+    function setBotOracle(address _botOracle) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+        botOracle = _botOracle;
+    }
+
+    /* External functions */
+
+    /**
+     * @notice Transfers to solana
+     * @param _nonce - nonce for wormhole
+     */
+    function transferToSolana(uint32 _nonce) external {
+        _requireIsInitialized();
+
+        require(msg.sender == governanceAddress, ERROR_ONLY_GOVERNANCE);
+
+        uint256 balance = audiusToken.balanceOf(address(this));
```

Comment:
This initially had a variable to allow partial transfers. However, I think there is no risk to transferring the full balance here, and there is not a lot of benefit in having the ability of partial transfers.

Reply:
i think we actually do want partial transfers fwiw so that we can have 'extra' funds here if we want - let's add this, since we can also just always transfer the whole thing if necessary

Reply:
nvm per offline discussion it actually makes sense to avoid getting txs queued up on this